pegasus/sites.v3/code.org/views/learn_carousel_block.haml

```diff
@@ -0,0 +1,31 @@
+- short_code = I18n.t(:"#{id}_shortcode")
+- short_url = "https://hourofcode.com/#{short_code}"
+- show_sw_blocks = id == "starwars" && (request.locale != "en-US" || mobile?)
+- if show_sw_blocks
+  - launch_code = "starwarsblocks"
+- else
+  - launch_code = id
+
+.container-fluid
+  .row-fluid
+    .courseblock-tall
+      %a{href: "http://studio.code.org/s/#{id}/reset"}
+        .imgspan
+          %img{src: "/shared/images/courses/logo_tall_#{id}.jpg", style: "width: 100%;"}
+        .clear
+        .heading
+          %h4{style: "font-size: 18px;"}= I18n.t(:"#{id}_name")
+        .clear
+        .smalltext
+          = I18n.t(:"#{id}_shortdescription")
```

Comment:
Now tested and added overflow/ellipsis rules. Had to change the button to fit below the links instead. I think it still works.